runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -99,13 +104,84 @@ func (q *MetricsViewComparison) Resolve(ctx context.Context, rt *runtime.Runtime
 		q.Where = convertFilterToExpression(q.Filter)
 	}
 
+	err = q.calculateMeasuresMeta()
+	if err != nil {
+		return err
+	}
+
 	if !isTimeRangeNil(q.ComparisonTimeRange) {
 		return q.executeComparisonToplist(ctx, olap, q.MetricsView, priority, q.ResolvedMVSecurity)
 	}
 
 	return q.executeToplist(ctx, olap, q.MetricsView, priority, q.ResolvedMVSecurity)
 }
 
+func (q *MetricsViewComparison) calculateMeasuresMeta() error {
+	compare := !isTimeRangeNil(q.ComparisonTimeRange)
+
+	if !compare && len(q.ComparisonMeasures) > 0 {
+		return fmt.Errorf("comparison measures are provided but comparison time range is not")
+	}
+
+	if len(q.ComparisonMeasures) == 0 && compare {
+		// backwards compatibility
+		q.ComparisonMeasures = make([]string, len(q.Measures))
+		for i, m := range q.Measures {
+			q.ComparisonMeasures[i] = m.Name
+		}
+	} else {
+		for _, cm := range q.ComparisonMeasures {
+			found := false
+			for _, m := range q.Measures {
+				if m.Name == cm {
+					found = true
+					break
+				}
+			}
+			if !found {
+				return fmt.Errorf("comparison measure '%s' is not present in the measures list", cm)
+			}
+		}
+	}
```

Comment:
This validation can also be pushed to after the `q.measuresMeta` is built and avoid the nested loop